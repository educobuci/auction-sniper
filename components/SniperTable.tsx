import { FC } from 'react'

export type SniperTableRow = {
  id: string
  lastPrice: string
  lastBid: string
  status: string
}

export type SniperTableProps = {
  rows: SniperTableRow[]
}

const TableCell = ({ children }) =>
  <td className="border border-gray-300 dark:border-gray-500 px-2 py-1">{children}</td>

const TableHeadCell = ({ children }) =>
  <th className="border border-gray-300 dark:border-gray-500 px-2 py-1">{children}</th>

const SniperTable: FC<SniperTableProps> = ({ rows }) =>
  <table>
    <thead>
      <tr>
        <TableHeadCell>Item</TableHeadCell>
        <TableHeadCell>Last Price</TableHeadCell>
        <TableHeadCell>Last Bid</TableHeadCell>
        <TableHeadCell>Status</TableHeadCell>
      </tr>
    </thead>
    <tbody>
      { rows.map(({ id, lastPrice, lastBid, status }) => 
        <tr key={id}>
          <TableCell>{id}</TableCell>
          <TableCell>{lastPrice}</TableCell>
          <TableCell>{lastBid}</TableCell>
          <TableCell>{status}</TableCell>
        </tr>
      )}
    </tbody>
  </table>
  
  
export default SniperTable