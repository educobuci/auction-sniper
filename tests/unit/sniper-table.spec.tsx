import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import SniperTable, { SniperTableRow } from '../../components/SniperTable'

describe('Sniper table', () => {
  const row: SniperTableRow = { id: 'id', lastPrice: '1000', lastBid: '1200', state: 'Bidding' }

  it('should have enough columns', () => {
    const { container } = render(<SniperTable rows={[row]} />)
    expect(container.querySelectorAll('td').length).toEqual(4)
  })

  it('should set sniper values in columns', () => {
    const { container } = render(<SniperTable rows={[row]} />)
    const cols = Array.from(container.querySelectorAll('td')).map(el => el.textContent)
    expect(cols).toEqual(Object.values(row))
  })

  it.skip('should set sniper values in columns [SNAPSHOT]', () => {
    const table = renderer.create(<SniperTable rows={[row]} />)
    expect(table.toJSON()).toMatchSnapshot()
  })
})