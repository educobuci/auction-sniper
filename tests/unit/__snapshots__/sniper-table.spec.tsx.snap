// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sniper table should set sniper values in columns [SNAPSHOT] 1`] = `
<table>
  <thead>
    <tr>
      <th
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        Item
      </th>
      <th
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        Last Price
      </th>
      <th
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        Last Bid
      </th>
      <th
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        Status
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        id
      </td>
      <td
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        1000
      </td>
      <td
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        1200
      </td>
      <td
        className="border border-gray-300 dark:border-gray-500 px-2 py-1"
      >
        Bidding
      </td>
    </tr>
  </tbody>
</table>
`;
